src/main/java/org/jenkinsci/plugins/workflow/cps/CpsStepContext.java

```diff
@@ -215,17 +215,12 @@ public String getDisplayName() {
 
     /**
      * Returns the thread that is executing this step.
-     * Needs to take {@link CpsThreadGroup} as a parameter to prove that the caller is in CpsVmThread.
      *
      * @return
      *      null if the thread has finished executing.
      */
     @CheckForNull CpsThread getThread(CpsThreadGroup g) {
-        CpsThread thread = g.threads.get(threadId);
-        if (thread == null) {
-            LOGGER.log(Level.FINE, "no thread " + threadId + " among " + g.threads.keySet(), new IllegalStateException());
-        }
-        return thread;
+        return g.getThread(threadId);
```

Comment:
Could just inline this method now I guess. No reason for it to exist as an instance method.

Reply:
Yeah, I looked into removing the method earlier, but ran into a [usage in `CpsBodyInvoker`](https://github.com/jenkinsci/workflow-cps-plugin/blob/e5e5f8054ff485685e0a5b36db65d549604ed956/src/main/java/org/jenkinsci/plugins/workflow/cps/CpsBodyInvoker.java#L151) that would require making `CpsStepContext.threadId` accessible and so it seemed like a wash.

Reply:
Ah, sorry, missed that this was encapsulating a `private` variable in this class too.